CONTRIBUTING.md

```diff
@@ -1,40 +1,118 @@
 
-Thanks! :heart: :heart: :heart:
+### Submitting A Bug Report
 
-The contribution guidelines are inspired by the Ruby on Rails contribution guidelines. 
+To report a bug, [open a new issue](https://github.com/iotaledger/iri/issues/new), and be sure to include as many details as possible, using the template.
 
-IRI Team
+**Note:** Minor changes such as fixing a typo can but do not need an open issue.
+
+If you also want to fix the bug, submit a [pull request](#pull-requests) and reference the issue.
+
+## Suggest a new feature
+
+This section guides you through suggesting a new feature. Following these guidelines helps maintainers and the community collaborate to find the best possible way forward with your suggestion.
+
+### Before suggesting a new feature
+
+**Ensure the feature has not already been suggested** by searching on GitHub under [**Issues**](https://github.com/iotaledger/iri/issues).
+
+### Suggesting a new feature
+
+To suggest a new feature, talk to the IOTA community and IOTA Foundation members in the #iri-discussion channel on [Discord](https://discord.iota.org/).
+
+If the IRI team approves your feature, the team will create an issue for it.
+
+## Build a new feature
+
+This section guides you through building a new feature. Following these guidelines helps give your feature the best chance of being approved and merged.
+
+### Before building a new feature
+
+Make sure to discuss the feature in the #iri-discussion channel on [Discord](https://discord.iota.org/).
+
+Otherwise, your feature may not be approved at all.
+
+### Building a new feature
+
+To build a new feature, check out a new branch based on the `dev` branch, and be sure to consider the following:
+
+- If the feature has a public facing API, make sure to document it, using Javadoc code comments
+
+- Where necessary, please write regression tests for your feature. Refer to our current [regression tests](https://github.com/iotaledger/iri/tree/dev/python-regression) for guidance.
+
+## Contribute to the IRI documentation
+
+The IRI documentation is hosted on https://docs.iota.org, which is built from content in the [documentation](https://github.com/iotaledger/documentation) repository.
+
+Please see the [guidelines](https://github.com/iotaledger/documentation/CONTRIBUTING.md) on the documentation repository for information on how to contribute to the documentation.
+
+## Pull requests
+
+This section guides you through submitting a pull request (PR). Following these guidelines helps give your PR the best chance of being approved and merged.
+
+### Before submitting a pull request
+
+When creating a pull request, please follow these steps to have your contribution considered by the maintainers:
+
+- A pull request should have exactly one concern (for example one feature or one bug). If a PR address more than one concern, it should be split into two or more PRs.
+
+- A pull request can be merged only if it references an open issue
```

Comment:
internal IF devs don't always follow this practice but I guess it is good to have it here